### ISSUE TITLE
Free Trials: Add notifications

### ISSUE THREAD

**Issue by mikeshelton1503** [Automattic/wp-calypso#1240]
We are going to utilize notifications to aid in the communication and engagement plan for free trials.

We should send 3 notifications during the trial period, one denoting the start of the trial, one denoting the end of the trial, and one (1) day before the trial expires.

#### Trial Start notification ####
_To be sent:_ just after sign up

![image](https://cloud.githubusercontent.com/assets/12596797/11573722/c205f3d4-99d5-11e5-892a-f739bf10d451.png)

Compact notification: "Your **WordPress.com Premium** trial just started!
Expanded notification: "Hooray! Your 14 day free trial of WordPress.com Premium on My Test Blog just started. [View the status of your trial.](#)"

#### Trial Ending notification ####
_To be sent:_ 24 hours before expiration

![image](https://cloud.githubusercontent.com/assets/12596797/11573734/d1347a92-99d5-11e5-8f8d-121db612976a.png)

Compact notification: "Your **WordPress.com Premium** trial ends tomorrow!"
Expanded notification: "Your free trial of WordPress.com Premium on My Test Blog ends tomorrow. Upgrade now to keep using all of the awesome [WordPress.com Premium features](#). // **[Upgrade Now]**"

#### Trial Expired notification ####
_To be sent:_ just after expiration

![image](https://cloud.githubusercontent.com/assets/12596797/11573608/31b18ece-99d5-11e5-901a-54ce5e1dd504.png)

Compact notification: "Your **WordPress.com Premium** trial just ended."
Expanded notification: "Your free trial of WordPress.com Premium on My Test Blog just ended. The plan's features have been removed from your account :(. // Don't worry, you can upgrade now to reclaim those features and your custom changes. Otherwise, thanks for giving it a try! // **[Upgrade Now]**

cc/ @danhauk @breezyskies for design and copy review

**Comment by mikeshelton1503** [comment]
Note: Currently other notification badges are displaying at 256x256px however that's a mistake. That's the 2x version. They should be displaying at 128x128px. The designs above use 128px badges for the expanded notifcations

**Comment by breezyskies** [comment]
Copy looks good! A couple notes:

- Should we mention the length of the trial in the _Trial Starting_ notification? 
- `awesome WordPress.com Premium features` in the _Trial Ending_ could link to the free trial hub, in case they aren't sure what those features are.
- In the _Trial Expired_ one, should we fade the icon out to the gray version?
- Should we modify the copy for any of these if they have used the domain credit?

**Comment by folletto** [comment]
Everything looks good to me. I like also the copy, even if it's almost too verbose. :)

My only doubt is...

> Trial Start notification

I get why this exists in terms of logic: one at the beginning, one warning, one at the end. However...

If the user just spend time reading, choosing and activating, is this notification really needed? Or it's because there's a sensible delay between the time the user activates and the trial starting? 

*If it's instant*, I see little value in this. We should probably delay it to a later moment with a different message. Maybe half-way through, sending a notification that says something along the lines of "I hope you're enjoying the trial! Have you tried all the features yet?". :)

**Comment by mikeshelton1503** [comment]
> I get why this exists in terms of logic: one at the beginning, one warning, one at the end.

You're exactly right here with the rationale for why I added it. It started with the "Trial Ending notification" which then seemed odd to only use for purposes of getting them to do what we wanted. So then I added the "Trial Expired notification" and that quickly felt weird to have an ending one but not a beginning one :). So while I the first one is less useful than others I felt that consistency in the way we use notifications was important.

The one thing is does so that is useful is it links to what we are calling the "Free Trial hub". It hasn't been designed yet (working on it today and tomorrow) but the general idea is it will be a "status box" at the top of the /plans page that displays: 1) that you're using a free trial 2) how many days left 3) a way to upgrade now and 4) a list of features included in the plan to try out. So there's value in using the notification show users where that is. @breezyskies and I discussed that the eventual best plan is to probably to use the "Free trial hub" as the thanks page but we agreed that's probably a change for v2.

>We should probably delay it to a later moment with a different message.

We also discussed having it be delayed until after the user navigates from the thank you page. So not instant though still fairly quickly after you sign-up.

**Comment by mikeshelton1503** [comment]
> Should we mention the length of the trial in the Trial Starting notification?

Yes, probably so. Done.

> awesome WordPress.com Premium features in the Trial Ending could link to the free trial hub, in case they aren't sure what those features are.

:+1: Done.

> In the Trial Expired one, should we fade the icon out to the gray version?

Agreed. Done.

> Should we modify the copy for any of these if they have used the domain credit?

I don't think that will affect these at all.

**Comment by breezyskies** [comment]
> > Should we modify the copy for any of these if they have used the domain credit?

> I don't think that will affect these at all.

My only consideration was warning the user about losing the domain if the trial expires.

**Comment by mikeshelton1503** [comment]
Yea we could do that. Maybe it's not until 1 day before the domain is going to be canceled and it says something like "yourdomain.com is set to be cancelled tomorrow." I think just that one additional one would be good, don't won't to overdo it with notifications. Though now I'll need to come up with some kind of badge for domains.

**Comment by mikeshelton1503** [comment]
> The one thing is does so that is useful is it links to what we are calling the "Free Trial hub". 

The Free Trial Hub: https://github.com/Automattic/wp-calypso/issues/1151#issuecomment-162624443

**Comment by rickybanister** [comment]
These look very beautiful, but I'm curious why the body copy is rendered in Merriweather, shouldn't we reserve that only for 'content'—things written by users rather than for 'interface' (which is what this copy is I think)?

**Comment by mikeshelton1503** [comment]
Need to update this flow to account for the 3 day extension. We will need to update `Trial  Ending` and we _may_ add one.

**Comment by mikeshelton1503** [comment]
> These look very beautiful, but I'm curious why the body copy is rendered in Merriweather, shouldn't we reserve that only for 'content'—things written by users rather than for 'interface' (which is what this copy is I think)?

@rickybanister Yes I thought that too however currently all notifications use Merriweather for the body copy (ex. https://cloudup.com/ct3skuyElRy) so that's a change we can open a separate issue for.

**Comment by fabianapsimoes** [comment]
As discussed with @mikeshelton1503, we're moving this to V2.

**Comment by drw158** [comment]
@mikeshelton1503 @rickybanister 

Good catch on the Merriweather. It is applied to to all content text (comments), so the styles are probably leaking to the achievement notification type.

As for the big badge image, it's technically not a mistake, it was intentional (in the CSS the images are a flexible width). It does look better small though. I'm in favor of making all of the detail view images 128px.

**Comment by fabianapsimoes** [comment]
Moved to T364-code, since this is a back-end change.
